env/types.go

```diff
@@ -125,20 +135,19 @@ func ValidIdentifier(n string) error {
 }
 
 // Set sets a variable.
+// Values are optionally parsed by a helper function if one is defined in
+// variableParseFunctions
 func Set(name, value string) error {
-	err := ValidIdentifier(name)
-	if err != nil {
+	if err := ValidIdentifier(name); err != nil {
 		return err
 	}
-	switch name {
-	case "QUIET":
-		if value == "" {
-			value = "on"
+	// parse and validate value if there's a function defined in variableParseFunctions
+	if parseFunction, ok := variableParseFunctions[name]; ok {
+		if parsedValue, err := parseFunction(value, name); err != nil {
+			return fmt.Errorf(text.FormatFieldInvalid, value, name)
```

Comment:
In this style the variable parsedValue is block scoped to the if/else so the else block is needed otherwise the variable goes out of scope and this causes a undefined variable compiler error.

Reply:
Yeah, you'd have to split up the assignment and the condition. Up to you, feel free to mark this comment as resolved.

Reply:
I think your proposal is a bit more idiomatic. Let me change it.